base_report_to_printer/ir_report.py

```diff
@@ -85,9 +85,32 @@ def print_direct(self, cr, uid, report_id, result, format, printer,
         'printing_action_ids': fields.one2many(
             'printing.report.xml.action', 'report_id', 'Actions',
             help='This field allows configuring action and printer on a per '
-                 'user basis'),
+                 'user (or company) basis'),
     }
 
+    def _get_company_id(self, cr, uid, context=None):
+        """
+        As it may happend that OpenERP force the uid to 1 to bypass rule (in
+        function field), we may sometimes read the price of the company of user
+        id 1 instead of the good one. Because we found the real uid and
+        company_id in the context in that case, I return this one. It also
+        allow other module to give the proper company_id in the context (like
+        it's done in product_standard_margin for example).
```

Comment:
Yes it is a copy (and not only for the docstring) from here: https://github.com/OCA/margin-analysis/blob/7.0/product_price_history/product_price_history.py#L224

Should I do something ?